ethereum/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -77,31 +94,54 @@ contract L1ERC20Bridge is IL1Bridge, IL1BridgeLegacy, AllowListed, ReentrancyGua
     /// @notice At the time of the function call, it is not yet deployed in L2, but knowledge of its address
     /// @notice is necessary for determining L2 token address by L1 address, see `l2TokenAddress(address)` function
     /// @param _governor Address which can change L2 token implementation and upgrade the bridge
-    /// @param _deployBridgeImplementationFee How much of the sent value should be allocated to deploying the L2 bridge
     /// implementation
-    /// @param _deployBridgeProxyFee How much of the sent value should be allocated to deploying the L2 bridge proxy
     function initialize(
         bytes[] calldata _factoryDeps,
         address _l2TokenBeacon,
-        address _governor,
-        uint256 _deployBridgeImplementationFee,
-        uint256 _deployBridgeProxyFee
+        address _governor
     ) external payable reentrancyGuardInitializer {
         require(_l2TokenBeacon != address(0), "nf");
         require(_governor != address(0), "nh");
         // We are expecting to see the exact three bytecodes that are needed to initialize the bridge
         require(_factoryDeps.length == 3, "mk");
         // The caller miscalculated deploy transactions fees
-        require(msg.value == _deployBridgeImplementationFee + _deployBridgeProxyFee, "fee");
         l2TokenProxyBytecodeHash = L2ContractHelper.hashL2Bytecode(_factoryDeps[2]);
         l2TokenBeacon = _l2TokenBeacon;
+        l2Governor = _governor;
+
+        factoryDepsHash = keccak256(abi.encode(_factoryDeps));
+    }
+
+    /// @dev Initializes a contract bridge for later use. Expected to be used in the proxy
+    /// @dev During initialization deploys L2 bridge counterpart as well as provides some factory deps for it
+    /// @param _factoryDeps A list of raw bytecodes that are needed for deployment of the L2 bridge
+    /// @notice _factoryDeps[0] == a raw bytecode of L2 bridge implementation
+    /// @notice _factoryDeps[1] == a raw bytecode of proxy that is used as L2 bridge
+    /// @notice _factoryDeps[2] == a raw bytecode of token proxy
+    /// @param _deployBridgeImplementationFee How much of the sent value should be allocated to deploying the L2 bridge
+    /// implementation
+    /// @param _deployBridgeProxyFee How much of the sent value should be allocated to deploying the L2 bridge proxy
+    function initializeChain(
+        uint256 _chainId,
+        bytes[] calldata _factoryDeps,
+        uint256 _deployBridgeImplementationFee,
+        uint256 _deployBridgeProxyFee
+    ) external payable reentrancyGuardInitializer {
+        // We are expecting to see the exact three bytecodes that are needed to initialize the bridge
+        require(_factoryDeps.length == 3, "mk1");
```

Comment:
NIT: May ignore it for now. 
Please use bigger revert messages for all new functionality.